src/utilities/_hypre_utilities.h

```diff
@@ -1599,10 +1610,10 @@ HYPRE_Real    hypre_cimag( HYPRE_Complex value );
 HYPRE_Complex hypre_csqrt( HYPRE_Complex value );
 #else
 #define hypre_conj(value)  value
-#define hypre_cabs(value)  fabs(value)
+#define hypre_cabs(value)  hypre_abs(value)
```

Comment:
If we leave this as `fabs` we get some type checking.  If we use `hypre_abs` then `hypre_cabs` would work for `int` as well.

Reply:
`hypre_abs` is a macro, so it works everywhere (CPU and GPU) and for everything(double, float, int, long, etc). `fabs` is a CPU function and available on GPUs as well, which accepts only `double`.

Reply:
I knew it was a macro, but I forgot that we need it for "single" and "quad" too (even though I talked about this very thing in my comment on `hypre_sqrt`).  I still wonder about the type checking question though.  Is that important to us?

Reply:
@falgout, what's your take on this one? I like the generic-ness of the macro, but I also like the safety of type checking. An alternative is to expressly define hypre_cabs to take <fabsf, fabs, or fabsl>, though this may have portability issues on GPUs.

Reply:
Just leave it as `hypre_abs` for now.  I'm not sure about the type checking question or the GPU side yet.